src/client.rs

```diff
@@ -334,8 +352,53 @@ impl<S: AsyncRead + AsyncWrite + Unpin + Send> Client<S> {
 
         self.connection.flush_stream().await?;
         let col_data = columns.iter().map(|c| format!("{}", c)).join(", ");
-        let query = format!("INSERT BULK {} ({})", table, col_data);
-
+        let mut query = format!("INSERT BULK {} ({})", table, col_data);
+        if options.bits() > 0 || order_hints.len() > 0 {
+            let mut add_separator = false;
+            query.push_str(" WITH (");
+            if options.contains(SqlBulkCopyOptions::KeepNulls) {
```

Comment:
Can't you use `bitflags`'s `Flag::iter` and `join` to simplify the whole separator business?